sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1065,7 +1031,7 @@ case class ScalaUDF(
        """.stripMargin)
   }
 
-  private[this] val converter = CatalystTypeConverters.createToCatalystConverter(dataType)
+  private[this] val resultConverter = CatalystTypeConverters.createToCatalystConverter(dataType)
 
   lazy val udfErrorMessage = {
```

Comment:
Seems can be just `val` now.

Reply:
With the tree node library, it's better to hide some computation via lazy val, so that we don't need to re-compute it every time when it gets transformed and makes a new copy.

Reply:
Ok. makes sense.